fe/src/main/java/org/apache/doris/qe/Coordinator.java

```diff
@@ -882,12 +889,60 @@ private void computeFragmentHosts() throws Exception {
         }
     }
 
-    //One fragment could only have one HashJoinNode
-    private boolean isColocateJoin(PlanNode node) {
-        if (Config.disable_colocate_join) {
+    private boolean needParallelInstance(PlanNode leftMostNode) {
+        if (getParallelExecInstanceNum() <= SessionVariable.MIN_EXEC_INSTANCE_NUM) {
             return false;
         }
 
+        if (leftMostNode instanceof OlapScanNode) {
+            OlapScanNode olapScanNode = (OlapScanNode) leftMostNode;
+
+            //case 1: the small table of broadcast join need not parallel
+            PlanFragment destFragment =  olapScanNode.getFragment().getDestFragment();
+            if (destFragment != null)  {
+                PlanNode destRootNode = destFragment.getPlanRoot();
+                if (destRootNode instanceof HashJoinNode) {
+                    HashJoinNode joinNode = (HashJoinNode) destRootNode;
+                    if (!joinNode.isShuffleJoin()) {
+                        LOG.debug("ScanNode {} for fragment {} need not parallel because of broadcast join",
+                        olapScanNode.getOlapTable().getName(), olapScanNode.getFragmentId());
+                        return false;
+                    }
+                }
```

Comment:
Does this works? If the fragment is agg(hash(scan1, scan2)), for scan2, the RootNode of `destFragment` is `agg`.

I think if it isn't easy to match this case, we can return true for it. 

For broadcast join, I think the bigger influence is that the outer child will be parallel, and making it would need more hash table for inner child. So it's ok to scan parallel here

Reply:
Yes. It's doesn't work for  agg(hash(scan1, scan2)).

So, should we parallel for all query? 
In my test, even if we parallel for small table scan, the performance loss is very little.

Reply:
For simple, I think we can parallel for all queries. 
However one thing we should notice is that when we enable fragment parallel, FE would send more `PlanFragmentInstance` than before, which would cost FE more CPU to serialize requests and more network bandwidth to transfer requests. Then as the number of concurrent queries increase, there would be some bottleneck for FE.

Reply:
> For simple, I think we can parallel for all queries.

OK.


> which would cost FE more CPU to serialize requests and more network bandwidth to transfer requests.

Yes, this is a side-effect for this improve. But which should be acceptable.